src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.cpp

```diff
@@ -40,104 +36,406 @@ namespace fastdds {
+    if (!request.type_ids().empty())
+    {
+        // Check if the received request has been done before and needed a continuation point
+        std::lock_guard<std::mutex> lock(requests_with_continuation_mutex_);
+        if (!request.continuation_point().empty())
+        {
+            auto requests_it = requests_with_continuation_.find(request.type_ids());
+            if (requests_it != requests_with_continuation_.end())
+            {
+                // Get the dependencies without chechking the registry
+                type_dependencies = requests_it->second;
+                type_dependencies_result = RETCODE_OK;
             }
-            case TypeLookup_getDependencies_HashId:
+        }
+        else
+        {
+            // Get the dependencies from the registry
+            type_dependencies_result =
+                    fastrtps::rtps::RTPSDomainImpl::get_instance()->type_object_registry_observer().
+                            get_type_dependencies(request.type_ids(), type_dependencies);
+
+            // If there are too many dependent types, store the type dependencies for future requests
+            if (type_dependencies_result == RETCODE_OK && type_dependencies.size() > MAX_DEPENDENCIES_PER_REPLY)
             {
-                const TypeLookup_getTypeDependencies_In in = request.data().getTypeDependencies();
-                TypeLookup_getTypeDependencies_Out out;
-                //for (size_t index = 0; index < in.type_ids.size(); ++index)
-                {
-                    // TODO (adelcampo) Change to xtypes
-                    // out.dependent_typeids = factory_->typelookup_get_type_dependencies(
-                    //     in.type_ids, in.continuation_point, out.continuation_point, 255); // TODO: Make configurable?
-                }
+                requests_with_continuation_.emplace(request.type_ids(), type_dependencies);
+            }
+        }
+    }
 
-                TypeLookup_Reply* reply = static_cast<TypeLookup_Reply*>(typelookup_manager_->reply_type_.createData());
-                TypeLookup_getTypeDependencies_Result result;
-                result.result(out);
-                reply->return_value().getTypeDependencies(result);
-                reply->header().relatedRequestId() = request.header().requestId();
+    if (RETCODE_OK == type_dependencies_result)
+    {
+        // Prepare and send the reply
+        TypeLookup_getTypeDependencies_Out out = prepare_get_type_dependencies_response(
+            request.type_ids(), type_dependencies, request.continuation_point());
+        answer_request(request_id, rpc::RemoteExceptionCode_t::REMOTE_EX_OK, out);
+    }
+    else
+    {
+        // If the type dependencies are not found, log error
+        EPROSIMA_LOG_WARNING(TYPELOOKUP_SERVICE_REQUEST_LISTENER, "Error getting type dependencies.");
+        answer_request(request_id, rpc::RemoteExceptionCode_t::REMOTE_EX_INVALID_ARGUMENT);
+    }
```

Comment:
Again, it is not the same if the parameter is invalid (non-direct hash TypeIdentifier) or if it is unknown.

Reply:
added better error check in [d054d39](https://github.com/eProsima/Fast-DDS/pull/4240/commits/d054d39484489f3890638c8cfe4fe4c023139c74)